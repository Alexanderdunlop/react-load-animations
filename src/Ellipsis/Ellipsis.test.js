import React from 'react'
import renderer from 'react-test-renderer'
import { Ellipsis } from '..'

it('renders correctly', () => {
  const tree = renderer.create(<Ellipsis />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with both props', () => {
  const tree = renderer.create(<Ellipsis width={500} height={500} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render with invalid props', () => {
  const tree = renderer
    .create(<Ellipsis width={'500'} height={'500'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
