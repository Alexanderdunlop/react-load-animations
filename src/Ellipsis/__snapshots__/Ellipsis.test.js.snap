// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render with invalid props 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="lds-ellipsis"
    style={
      Object {
        "height": "500",
        "width": "500",
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>,
]
`;

exports[`renders correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="lds-ellipsis"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>,
]
`;

exports[`renders correctly with both props 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="lds-ellipsis"
    style={
      Object {
        "height": 500,
        "width": 500,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>,
]
`;
